test/unit/lib/runner/test-runner/test-report-handler-tests.js

```diff
@@ -429,43 +436,200 @@ describe('lib/runner/test-runner/test-report-handler.js', function() {
+    let Emitter;
+    let configHandler;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/test-runner/test-report-handler.js');
+
+      Emitter = {
+        mixIn: function(myObject) {
+            myObject.on = sinon.stub();
+            myObject.emit = sinon.stub();
+        },
+      };
+      sinon.spy(Emitter, 'mixIn');
+
+      configHandler = {
+        get: sinon.stub(),
+      };
+
+      mockery.registerMock('../../util/emitter.js', Emitter);
+      mockery.registerMock('../runner-event-dispatch/runner-event-dispatch.js', {});
+      mockery.registerMock('../../util/config-handler.js', configHandler);
+
+      testReportHandler = require('../../../../../lib/runner/test-runner/test-report-handler.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    it('should call configHandler.get twice', function() {
+      testReportHandler._handleTestReportSummary();
+
+      expect(configHandler.get.callCount).to.equal(2);
+    });
+
+    it('should configHandler.get with \'reporter\'', function() {
+      testReportHandler._handleTestReportSummary();
+
+      expect(configHandler.get.args[0]).to.deep.equal(['reporter']);
+    });
+
+    describe('if configHandler.get returns \'basic\'', function() {
+      it('should call testReportHandler.emit with the event \'testReportHandler.testReportReadyToPrint\'' +
+        ' \'basic\', and the ._report', function() {
+        configHandler.get.returns('basic');
+
+        testReportHandler._handleTestReportSummary();
+
+        expect(testReportHandler.emit.args).to.deep.equal([
+          [
+            'testReportHandler.testReportSummaryReadyToPrint',
+            'basic',
+            testReportHandler._report,
+          ],
+        ]);
+      });
+    });
+
+    it('should call configHandler.get with  \'reportPath\'', function() {
+      testReportHandler._handleTestReportSummary();
+
+      expect(configHandler.get.args[1]).to.deep.equal(['reportPath']);
+    });
+
+    describe('if configHandler.get(\'reportPath\') returns a value', function() {
+      it('should call configHandler.get a total of 3 times', function() {
+        configHandler.get.returns(true);
+
+        testReportHandler._handleTestReportSummary();
+
+        expect(configHandler.get.callCount).to.equal(3);
+      });
+
+      it('should call configHandler.get with \'reportFormat\'', function() {
+        configHandler.get.returns(true);
+
+        testReportHandler._handleTestReportSummary();
+
+        expect(configHandler.get.args[2]).to.deep.equal(['reportFormat']);
+      });
+
+      describe('if configHandler.get(\'reportFormat\') reutnrs \'JSON\'', function() {
+        it('should call testReportHandler.emit with the event \'testReportHandler.testReportReadyToPrint\'' +
+          ' \'basic\', and the report', function() {
```

Comment:
I see in the assertion below that the second parameters is `JSON` and not `basic` as specified in the description here. Is there some sort of disconnect here?